crates/consensus/auto-seal/src/lib.rs

```diff
@@ -115,13 +115,13 @@ where
         pool: Pool,
         to_engine: UnboundedSender<BeaconEngineMessage>,
         canon_state_notification: CanonStateNotificationSender,
+        mode: MiningMode,
```

Comment:
this is the best solution imo, even if it makes calling it slightly more verbose